placement/algo/sharded_helper.go

```diff
@@ -848,7 +851,7 @@ func markAllShardsAvailable(
 	for _, instance := range p.Instances() {
 		for _, s := range instance.Shards().All() {
 			if s.State() == shard.Initializing {
-				p, err = markShardAvailable(p, instance.ID(), s.ID(), opts)
+				p, err = markShardsAvailable(p, instance.ID(), []uint32{s.ID()}, opts)
```

Comment:
can you do something like
```
p, err = markShardsAvailable(p, instance.ID(), instance.Shards().ShardsForState(shard.Initializing), opts) 
```
here and save the iteration?

Reply:
@cw9 hm I tried that but getting some strange test errors.  Can you help me out here (I'm travelling the next day and a bit):
```

Reply:
talked offline: going to leave as is because the `markShardsAvailable` method expects `[]uint32` and `ShardsForState(..)` returns `[]Shard`, so we'd have to do the extra iteration to convert anyway.